src/constants/other/data-helpers-other.js

```diff
@@ -302,3 +305,132 @@ export function generateSeriesForClock(options) {
+  // statistics calculation
+  const min = sortedData[0];
+  const q1 = lastIteration ? oldQ1 : getMedianOfDataSet(sortedData.slice(0, leftMedIndex));
+  const med = lastIteration ? oldMed : getMedianOfDataSet(sortedData);
+  const q3 = lastIteration ? oldQ3 : getMedianOfDataSet(sortedData.slice(rightMedIndex, sortedData.length));
+  const max = sortedData[sortedData.length - 1];
+  // outliers helpers
+  const acceptedDistance = (q3 - q1) * 1.5;
+  const acceptableMin = q1 - acceptedDistance;
+  const acceptableMax = q3 + acceptedDistance;
+  // skip outliers if not neccesary to apply them
+  if (!applyOutliers || lastIteration || (acceptableMin < min && acceptableMax > max)) {
+    statistics.push(limitNumberOfDecimals(min));
+    statistics.push(limitNumberOfDecimals(q1));
+    statistics.push(limitNumberOfDecimals(med));
+    statistics.push(limitNumberOfDecimals(q3));
+    statistics.push(limitNumberOfDecimals(max));
+
+    result.statistics = statistics;
+    result.outliers = (outliers || []).map(value => [target, limitNumberOfDecimals(value)]);
+    return result;
+  }
+  const rawOutliers = [];
+  const filteredData = sortedData.filter((point) => {
+    if (point > acceptableMin &&  point < acceptableMax) {
+      return true;
+    }
+    rawOutliers.push(point);
+    return false;
+  });
+  const params = { lastIteration: true, outliers: rawOutliers, oldQ1: q1, oldMed: med, oldQ3: q3 };
+  return calculateStatistics(filteredData, target, applyOutliers, params);
+}
+
+export function generateInitialDataBoxplot(options) {
+  const { min, max } = options;
+  const NUMBER_OF_BOXPLOTS = 4;
+  const result = {};
+  for (let i = 0; i < NUMBER_OF_BOXPLOTS; i += 1) {
+    result[i + 1] = generateRandomPointsBoxplot(min, max, 5);
+  }
+  return result;
+}
+
+export function generateBoxplotSeries(data, options) {
+  const result = {};
+  Object.keys(data).forEach((boxplot) => {
+    result[boxplot] = calculateStatistics(data[boxplot], Number(boxplot) - 1, options.outliers);
+  });
+  let outliers = [];
+  Object.keys(result).forEach(boxplot => outliers = outliers.concat(result[boxplot].outliers));
+
+  return [{
+    name: 'Statistics',
+    data: Object.keys(result).map(boxplot => result[boxplot].statistics),
+    tooltip: {
+      headerFormat: '<em>Boxplot {point.key}</em><br/>'
+    }
+  }, {
+    name: 'Outliers',
+    color: Highcharts.getOptions().colors[0],
+    type: 'scatter',
+    data: outliers,
+    marker: {
+      fillColor: 'white',
+      lineWidth: 2,
+      lineColor: Highcharts.getOptions().colors[0]
+    },
+    tooltip: {
+      pointFormat: 'Value: {point.y}'
+    }
+  }];
+}
+
+export function averageLineBoxplot(data) {
+  let sum = 0;
+  let count = 0;
+  Object.keys(data).forEach((boxplot) => {
+    data[boxplot].forEach((point) => {
+      sum += point;
+      count += 1;
+    });
+  });
+
+  if (count > 0) {
+    const average = Math.round(sum * 100 / count) / 100;
```

Comment:
In the future all these things should be changed to helper function limitNumberOfDecimals